src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -22,7 +22,7 @@ protected internal JsonConverter()
             // In the future, this will be check for !IsSealed (and excluding value types).
             CanBePolymorphic = TypeToConvert == JsonTypeInfo.ObjectType;
             IsValueType = TypeToConvert.IsValueType;
-            CanBeNull = !IsValueType || TypeToConvert.IsNullableOfT();
+            CanBeNull = default(T) is null;
```

Comment:
I did verify this is faster for value types (same as for reference types).

Reply:
It seems like other usages of `TypeToConvert.IsNullableOfT` and `CanBeNull` should be changed or removed.

Reply:
It's still being used by the nullable converter factory and in a few reflection contexts where the generic type is not known statically.

Reply:
I've decided to keep `CanBeNull` in the few places that are not part of the hot path, since it's probably easier to parse compared to `default(T) is null`.